crates/rooch-types/src/authentication_key.rs

```diff
@@ -14,6 +15,26 @@ use std::{fmt::Display, str::FromStr};
 #[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
 pub struct AuthenticationKey(#[serde_as(as = "Readable<Hex,_>")] Vec<u8>);
 
+/// Define authentication key type for each blockchain
+#[derive(Debug, Clone, Copy, PartialEq, Eq, ArgEnum)]
+pub enum AuthenticationKeyType {
+    P2PKH,
+    P2SH,
+}
```

Comment:
AuthenticationKey does not need a `type`. It is just bytes. `P2PKH` and `P2SH` should be the Bitcoin address type?

Reply:
Yes. Since we have converted payload bytes to an address as an authentication key, I do think there should be an authentication key type representing the address type, e.g. Bitcoin address.

Reply:
I originally wrote AddressType but when it is being used in `Update.rs`, then we have address and address-type flags. But the address-type flag only means not the Rooch address type but the address contained in the authentication key, so I changed address type to authentication key type as the name.

Reply:
I think bitcoin-address-type is more appropriate. So I will refactor it for now.

Reply:
Maybe we should not support AuthenticationKey and rotate_authentication_key for the Eth validator and Bitcoin validator. We just verified the signature with the sender's multi-chain address. 

```
assert!(sender_address == address in the signature).
```

Because Eth and Bitcoin do not support rotate_authentication_key, If we support it, it will confuse users.

Reply:
> Maybe we should not support AuthenticationKey and rotate_authentication_key for the Eth validator and Bitcoin validator. We just verified the signature with the sender's multi-chain address.
> 
> ```
> assert!(sender_address == address in the signature).
> ```
> 
> Because Eth and Bitcoin do not support rotate_authentication_key, If we support it, it will confuse users.

I understand that Rooch address is using blake2b256 to hash bytes as authentication keys, but in the Ethereum and Bitcoin contexts, it is directly used address as authentication keys. Maybe we can use the pre-context of the address, such as base58 bytes as the authentication keys and users can opt in to encode the base58 bytes to an bitcoin address. I think that way to go is okay.

Alternatively, we can also remove the authentication key and rotate authentication key for ETH and BTC as you described it's not needed to have an address as the authentication key.

So, in two ways, we can refactor it:

1. Support pre-context bytes as the authentication key, like Rooch is using blake2b256 hash as the authentication key. Similarly, Bitcoin can use pre-context bytes as authentication key that can be encoded by base58 to form a Bitcoin address.
2. Remove authentication keys for ETH and Bitcoin.

Reply:
> Maybe we should not support AuthenticationKey and rotate_authentication_key for the Eth validator and Bitcoin validator. We just verified the signature with the sender's multi-chain address.
> 
> ```
> assert!(sender_address == address in the signature).
> ```
> 
> Because Eth and Bitcoin do not support rotate_authentication_key, If we support it, it will confuse users.

The prerequisite is that we can recover the signature to an address. This is applied for Ethereum but not Bitcoin. Because Ethereum is using 65-bytes signature and the last bytes can be used to recover to an address. Different blockchains have different algorithms and implementations. We should unify them as one.

Reply:
It is hard to unify. We follow the origin chain's algorithms and implementations.

Reply:
> It is hard to unify. We follow the origin chain's algorithms and implementations.

So what's your conclusion? Remove Ethereum's rotate authentication keys function and only recover the signature to an ETH address and compare it with the sender, while keeping Bitcoin's rotate authentication keys unchanged?

Reply:
I merge this PR first, we can discuss this later. The Eth validator's scenarios is for support Eth transaction and Eth wallet(eg. MetaMask), but the Bitcoin validator's scenarios not clear, maybe we can make it TBD.